base_classes/nyaml/NXsample.yaml

```diff
@@ -0,0 +1,365 @@
+  unit_cell_class:
+    doc: |
+      In case it is all we know and we want to record/document it
+    enumeration: [triclinic, monoclinic, orthorhombic, tetragonal, rhombohedral, hexagonal, cubic]
+  space_group:
+    doc: |
+      Crystallographic space group
+    dimensions:
+      dim: [[1, n_comp]]
+  point_group:
+    doc: |
+      Crystallographic point group, deprecated if space_group present
+    dimensions:
+      dim: [[1, n_comp]]
+  path_length(NX_FLOAT):
+    unit: NX_LENGTH
+    doc: |
+      Path length through sample/can for simple case when
+      it does not vary with scattering direction
+  path_length_window(NX_FLOAT):
+    unit: NX_LENGTH
+    doc: |
+      Thickness of a beam entry/exit window on the can (mm)
+      - assumed same for entry and exit
+  thickness(NX_FLOAT):
+    unit: NX_LENGTH
+    doc: |
+      sample thickness
+  transmission(NXdata):
+    doc: |
+      As a function of Wavelength
+  temperature_log(NXlog):
+    deprecated: |
+      use ``temperature``, see: https://github.com/nexusformat/definitions/issues/816
+    doc: |
+      temperature_log.value is a link to e.g. temperature_env.sensor1.value_log.value
+  temperature_env(NXenvironment):
+    doc: |
+      Additional sample temperature environment information
+  magnetic_field(NXlog):
+    doc: |
+      magnetic_field.value is a link to e.g. magnetic_field_env.sensor1.value
+  magnetic_field_log(NXlog):
+    deprecated: |
+      use ``magnetic_field``, see: https://github.com/nexusformat/definitions/issues/816
+    doc: |
+      magnetic_field_log.value is a link to e.g. magnetic_field_env.sensor1.value_log.value
+  magnetic_field_env(NXenvironment):
+    doc: |
+      Additional sample magnetic environment information
+  external_DAC(NX_FLOAT):
+    unit: NX_ANY
+    doc: |
+      value sent to user's sample setup
+  external_ADC(NXlog):
+    doc: |
+      logged value (or logic state) read from user's setup
+  short_title:
+    doc: |
+      20 character fixed length sample description for legends
+    
+    # How is the string length limitation imposed by the XSD?
+  rotation_angle(NX_FLOAT):
+    unit: NX_ANGLE
+    doc: |
+      Optional rotation angle for the case when the powder diagram has
+      been obtained through an omega-2theta scan like from a traditional
+      single detector powder diffractometer.
+      Note, it is recommended to use NXtransformations instead.
+  x_translation(NX_FLOAT):
+    unit: NX_LENGTH
+    doc: |
+      Translation of the sample along the X-direction of the laboratory coordinate system
+      Note, it is recommended to use NXtransformations instead.
+  distance(NX_FLOAT):
+    unit: NX_LENGTH
+    doc: |
+      Translation of the sample along the Z-direction of the laboratory coordinate system.
+      Note, it is recommended to use NXtransformations instead.
+  (NXpositioner):
+    doc: |
+      Any positioner (motor, PZT, ...) used to locate the sample
+  (NXoff_geometry):
+    exists: ['min', '0']
```

Comment:
Its a base class so optional by default

Reply:
I guess the idea was to have at least one of them by default, but then it should be named, right?
Like: geometry(NXoff_geometry)

Reply:
I understand the intention and fully agree technically, however, as far as I recall these constraints on base classes currently take/have no effect instead they would have to be defined in an application definition which uses an instance of this base class

Reply:
Ok, I commented it out for now